certbot/storage.py

```diff
@@ -256,8 +259,26 @@ def __init__(self, config_filename, cli_config):
         self.live_dir = os.path.dirname(self.cert)
 
         self._fix_symlinks()
+        if update_symlinks:
+            self._update_symlinks()
         self._check_symlinks()
 
+    def __str__(self):
+        """Returns a string representation of the object."""
+        # import ipdb; ipdb.set_trace()
+        outdict = {"Lineage": self.lineagename,
+                   "Domains": " ".join(self.names())}
+        return "RenewableCert(%s)" % str(outdict)
```

Comment:
> grabbed the date formatting from openssl x509 -inform pem -noout -text -in

Huh! That's wacky of the OpenSSL project...

> what's a possible use case for machine-readable?

People writing their own renewal scripts (maybe because they have criteria or associated actions that aren't well-covered by the behavior of `certbot renew`) or writing pre- and post-hooks for renewal that need to know more about what certificates exist, again because they want to take some action based on what happens or what exists. (For example, maybe something about notification to a human administrator, or something about synchronization between machines in a cluster -- like rsyncing only the certificates that changed in the course of the renewal process, perhaps?)

> can we default to human-readable, and then also have a machine-readable flag?

Yep, I think that would be a good choice.

> I really like the (89 days from now / EXPIRED) idea!

There are probably a lot of library approaches to doing this -- for example if you have a pair of `datetime` objects you can subtract them and then refer to the `.days` attribute of the resulting `timedelta` objects, but if you can get the expiry time as Unix time, you can say something like

```
diff = expiry - time.now()
if diff <= 0:
    return "EXPIRED"
elif diff < 86400:
    return "expiring today"
elif diff < 86400*2:
    return "1 day"
else:
    return "{0} days".format(diff % 86400)
```